LiveStream/ViewModels/LiveStreamViewModel.swift

```diff
@@ -396,3 +528,60 @@ private func startDateMoreThanFifteenMinutesAgo(event: LiveStreamEvent) -> Bool
 private func didEndNormally(event: LiveStreamEvent) -> Bool {
   return !event.liveNow && event.definitelyHasReplay
 }
+
+private enum TimeIntervalBufferEvent<T> {
```

Comment:
This is a pretty cool state machine approach that someone in the ReactiveSwift slack channel suggested to use for buffering on an interval. I just made a few small changes for our purposes. It may be overkill and perhaps we'd prefer that this is an operator in our ReactiveExtensions but it works!